### DISCUSSION TITLE
openbb.stocks.search API is slow

### DISCUSSION THREAD

**Discussion by ori-maci** [OpenBB-finance/OpenBB#4560] [General]
Hello I am wondering if anyone else is facing slow response issues with this API in the SDK: 

`companies_response = openbb.stocks.search(query=search_term, limit=1)`   (on average takes ~3.25 seconds but sometimes 2-3x longer) 

It seems that the `limit` does not improve the response time of the search. 

Question 1: Is there a plan to improve the performance of the search? 
Question 2: I am noticing that `limit` is not being honored. Is that intentional (Example: Search for `bo`)

**Comment by jmaslek** [comment]
So unfortunately the speed will likely not be improved upon as we are just using the finance_database to grab tickers.  We may be able to implement certain caching that could _potentially_ speed up the loading, but the search function will still take the majority of the time.

As for the limit functionality, this is an area of improvement on our end.  The search function returns a df as well as prints a table with `limit` number of rows.  if you assign the result to a dataframe, there will still print out a result.

![Screenshot 2023-03-22 at 9 54 21 AM](https://user-images.githubusercontent.com/18151143/226926225-1170a015-b050-432a-b0e0-235f36c29807.png)
